lib/counter_culture.rb

```diff
@@ -286,7 +286,11 @@ def change_counter_cache(options)
       if id_to_change && options[:counter_column]
         delta_magnitude = if options[:delta_column]
                             delta_attr_name = options[:was] ? "#{options[:delta_column]}_was" : options[:delta_column]
-                            self.send(delta_attr_name) || 0
+                            val = self.send(delta_attr_name)
+                            # check if value is a string. This can happen if the
+                            # delta_column is an enum
+                            val = val.is_a?(String) ? self.class.send(options[:delta_column].pluralize)[val] : val
```

Comment:
In the issue I opened I still had the `enumerize` gem installed, that's why there was this special `Enumerize::Value` object. I realized this when writing the unit test and the test failed to find the object Enumerize.

I feel the same as you.

Sadly, Rails does not really provide ways to actually detect if the column is an enum. It just adds a bunch of methods that allow you to work with it, see here: http://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html

So what we could do to make a better guess is:

* Check if self.class.send(options[:delta_column].pluralize) returns a Hash

Any further ideas?